weave/trace_server_bindings/stainless_trace_server.py

```diff
@@ -0,0 +1,838 @@
+"""
+This file exists as a drop in replacement for remote_http_trace_server.py
+
+It mirrors the same API to simplify migration.  In future, we will remove remote_http_trace_server.py altogether.
+"""
+
+import logging
+from collections.abc import Iterator
+from typing import Any, Optional, Union, cast
+
+logger = logging.getLogger(__name__)
+
+try:
+    from weave_server_sdk import WeaveTrace
+except ImportError:
+    WEAVE_SERVER_SDK_AVAILABLE = False
+else:
+    WEAVE_SERVER_SDK_AVAILABLE = True
+
+from weave.trace.env import weave_trace_server_url
+from weave.trace.settings import max_calls_queue_size, should_enable_disk_fallback
+from weave.trace_server import trace_server_interface as tsi
+from weave.trace_server_bindings.async_batch_processor import AsyncBatchProcessor
+from weave.trace_server_bindings.http_utils import log_dropped_call_batch
+from weave.trace_server_bindings.models import (
+    Batch,
+    EndBatchItem,
+    ServerInfoRes,
+    StartBatchItem,
+)
+from weave.utils.retry import _is_retryable_exception, get_current_retry_id, with_retry
+from weave.wandb_interface import project_creator
+
+REMOTE_REQUEST_BYTES_LIMIT = (
+    (32 - 1) * 1024 * 1024
+)  # 32 MiB (real limit) - 1 MiB (buffer)
+
+
+class RemoteHTTPTraceServer(tsi.TraceServerInterface):
```

Comment:
There seems to be a lot of duplicated logic with the baseline class. Can we subclass and just override the parts that differ, e.g. setting `_client` to `WeaveTrace`...

Reply:
The interface is just a protocol, so you have to implement everything anyways.  

The implementation of `stainless_trace_server.py` is intended to look like https://github.com/wandb/weave/blame/andrew/stainless-bindings/weave/trace_server_bindings/remote_http_trace_server.py#L39, just using the Stainless client under the hood

Reply:
We could refactor the `RemoteHTTPTraceServer` to share code with this, but seeing as the intent is to remove it in the near future I don't think it's necessary